extensions/spatial.sigmf-ext.md

```diff
@@ -0,0 +1,372 @@
+direction of the positive X-axis. Based on this, as an example, a horizontal
+uniform linear array aperture would be defined along the Y-axis (see examples
+section below).
+
+## 0 Datatypes
+
+This extension defines the following datatypes:
+
+|type|long-form name|description|
+|----|--------------|-----------|
+|bearing|signal direction bearing|Quantitative representation of a direction with optional error.|
+|cartesian_point|cartesian position|Cartesian coordinate triplet [x, y, z]|
+
+### 0.1 The `bearing` Object
+
+The `bearing` field is used to describe relative one or two dimensional signal
+directions. The units are always in degrees; while it is legal for this field to
+have any value it is RECOMMENDED that it be between 0 and 360 or +/- 180.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`azimuth`|false|float|degrees|Azimuth component of the direction in degrees increasing clockwise.|
+|`elevation`|false|float|degrees|Elevation component of the direction in degrees above horizon.|
+|`range`|false|float|meters|Line-of-sight slant range to emitter, if known, in meters.|
+|`az_error`|false|float|degrees|Error or uncertainty in the azimuth component.|
+|`el_error`|false|float|degrees|Error or uncertainty in the elevation component.|
+|`range_error`|false|float|meters|Error or uncertainty in the range component.|
+
+The `az_error`, `el_error`, and `range_error` field units are degrees, but the
+exact meaning of 'error' in this context is not explicitly defined. Applications
+SHOULD specify their specific meaning, and in general this should be interpreted
+as an uncertainty range. The error fields should be SHOULD NOT be used if the
+corresponding field estimates are not included.
+
+```json
+  "bearing": {
+    "azimuth": 211.2,
+    "elevation": 15.0,
+    "range": 30,
+    "az_error": 0.1,
+    "el_error": 0.1,
+    "range_error": 0.1
+  }
+```
+
+### 0.2 The `cartesian_point` Object
+
+The `cartesian_point` field is used to describe a single point in the Cartesian
+coordinate reference system. This object is necessary to define the phase center
+geometries for multidimensional arrays, and other cartesian locations.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`point`|true|float array|meters|Cartesian Point defined by a element array of [x,y,z] referenced to the `spatial` CRS.|
+
+
+## 1 Global
+
+The `spatial` extension adds the following fields to the `global` SigMF object:
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`num_elements`|true|int|N/A|Defines the number of phase centers / channels collected in the Collection or multichannel Dataset.|
+|`channel_num`|true|int|N/A|The channel number, shares index with `element_geometry`.|
+
+The number of elements MUST be defined here and is constant for a given
+Collection. It may be tempting to use the `core:num_channels` field however
```

Comment:
Assuming this field remains, would it make more sense to change "num" to "index"?

Maybe we should define a new convention for SigMF where "num" is used to count things and "index" is used to refer to a specific one?

Reply:
Yes, I think channel_index is more appropriate.